python/py-selenium/Portfile

```diff
@@ -4,31 +4,53 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-selenium
-version             3.141.0
+version             4.1.0
 license             Apache-2
 maintainers         {dstrubbe @dstrubbe} openmaintainer
 platforms           darwin
 
 description         Python language bindings for Selenium WebDriver
-long_description \
-    The selenium package is used to automate web browser interaction \
-    from Python. Several browsers/drivers are supported (Firefox, \
-    Chrome, Internet Explorer), as well as the Remote protocol.
+long_description    The selenium package is used to automate web \
+                    browser interaction from Python. Several \
+                    browsers/drivers are supported (Firefox, Chrome, \
+                    Internet Explorer), as well as the Remote protocol.
 
 homepage            https://www.seleniumhq.org/
-master_sites        pypi:s/${python.rootname}/
-distname            ${python.rootname}-${version}
 
-python.versions     27 37 38 39 310
-
-checksums           rmd160  6175bf1db6e8de270d0c34b9c58649ab1b118bc7 \
-                    sha256  deaf32b60ad91a4611b98d8002757f29e6f2c2d5fcaf202e1c9ad06d6772300d \
-                    size    854669
+python.versions     37 38 39 310
 
 if {${name} ne ${subport}} {
     depends_build-append \
-                    port:py${python.version}-setuptools
+                    port:py${python.version}-pip \
+                    port:py${python.version}-wheel
+
     depends_run-append \
                     port:py${python.version}-urllib3
+
+    master_sites    https://files.pythonhosted.org/packages/58/76/705b5c776f783d1ba7c630347463d4ae323282bbd859a8e9420c7ff79581/
+    distname        selenium-4.1.0-py3-none-any
```

Comment:
here you should/could use `${version}` instead of hardcoding the value, as also reported by `port lint`

Reply:
Thanks, done.

BTW, is there some clever way of automatically getting the hashed URL of pypi distributions like this one? See https://pypi.org/project/selenium/#files → https://files.pythonhosted.org/packages/58/76/705b5c776f783d1ba7c630347463d4ae323282bbd859a8e9420c7ff79581/selenium-4.1.0-py3-none-any.whl.

Reply:
You are not supposed to need to use these URLs that change with every version. The python portgroup is supposed to give you a working master_sites URL automatically. Why didn't it in this case?

Reply:
I think is has something to do with the `.whl` files and have seen this before as well but admittedly didn't spend much time to try and investigate it since it only affected a very small number of ports. I'm not sure if it can be fixed in the `python` PG but we'll not know unless someone will take a closer look - I can try to do so at some point.